zerver/views/streams.py

```diff
@@ -272,10 +313,17 @@ def update_stream_backend(
         if not user_profile.is_realm_owner:
             raise OrganizationOwnerRequired()
         user_profile.realm.ensure_not_on_limited_plan()
-        message_retention_days_value = parse_message_retention_days(
+        old_message_retention_days_value = stream.message_retention_days
+        new_message_retention_days_value = parse_message_retention_days(
             message_retention_days, Stream.MESSAGE_RETENTION_SPECIAL_VALUES_MAP
         )
-        do_change_stream_message_retention_days(stream, message_retention_days_value)
+        do_change_stream_message_retention_days(stream, new_message_retention_days_value)
+        send_change_stream_message_retention_days_notification(
+            user_profile,
+            stream,
+            old_value=old_message_retention_days_value,
+            new_value=new_message_retention_days_value,
+        )
```

Comment:
This should ideally be done in `do_change_stream_message_retention_days`, so that things are correct if we every call that from another code path.

We should pass `acting_user` into that function and change it to do a `RealmAuditLog` entry while we're at it; it looks like `do_rename_stream` has some sample code you can crib from.